### PR TITLE
ci: speedup free disk space job

### PR DESCRIPTION
**Pull request by xc2**
## Summary

Refer to https://github.com/jlumbroso/free-disk-space/pull/26 for more

## Checklist

- [x] Tests updated (or **not required**).
- [x] Documentation updated (or **not required**).

### CODE REVIEW

**Comment by chenjiahan** [comment]
Nice 👍🏻 Can you see how much faster Rspack's CI will be?

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -113,7 +113,7 @@ jobs:
       # Fix: Resolve disk space error "ENOSPC: no space left on device" on GitHub Actions runners
       - name: Free disk cache
         if: ${{ startsWith(runner.name, 'GitHub Actions') && inputs.target == 'x86_64-unknown-linux-gnu' && steps.check_cache.outputs.exists != 'true' && !inputs.skipable }}
-        uses: jlumbroso/free-disk-space@main
+        uses: xc2/free-disk-space@main
```
Comment (chenjiahan) [inline .github/workflows/reusable-build.yml:116]: I prefer pin to a release tag instead of `main`, `main` is unstable

**Comment by xc2** [comment]
> Nice 👍🏻 Can you see how much faster Rspack's CI will be?

We expect a reduction of ~1min regardless of the repo.

The actual execution time varies based on the availability of CPU and IO resources.

File: .github/workflows/reusable-build.yml (hunk 1 above)
Comment (xc2) [reply .github/workflows/reusable-build.yml]: DONE

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
The `Test Linux / Test Node 18` CI failed, but I think it is unrelated to this change...

**Comment by xc2** [comment]
> The `Test Linux / Test Node 18` CI failed, but I think it is unrelated to this change...

Awaiting https://github.com/web-infra-dev/rspack/pull/7240 to be approved or suggested a better solution.

**Review by chenjiahan — APPROVED** [review]
🎉

**Pull request merged**